clients/nodejs/validator-registration.js

```diff
@@ -0,0 +1,338 @@
+
+    $.client = new Nimiq.Client(clientConfig, $.consensus);
+    $.blockchain = $.consensus.blockchain;
+    $.accounts = $.blockchain.accounts;
+    $.mempool = $.consensus.mempool;
+    $.network = $.consensus.network;
+
+    Nimiq.Log.i(TAG, `Peer address: ${networkConfig.peerAddress.toString()} - public key: ${networkConfig.keyPair.publicKey.toHex()}`);
+
+    const chainHeight = await $.client.getHeadHeight();
+    const chainHeadHash = await $.client.getHeadHash();
+    Nimiq.Log.i(TAG, `Blockchain state: height=${chainHeight}, headHash=${chainHeadHash}`);
+
+    // This is the hardcoded address dedicated to validator registration
+    const recipientAddr = Nimiq.Address.fromUserFriendlyAddress('NQ07 0000 0000 0000 0000 0000 0000 0000 0000');
+
+    // Extract the validator configuration (we read the private keys from the config file)
+    let validatorPrivateKey = Nimiq.PrivateKey.unserialize(Nimiq.BufferUtils.fromHex(validator_config.ValidatorAccount.PrivateKey));
+    let signingPrivateKey = Nimiq.PrivateKey.unserialize(Nimiq.BufferUtils.fromHex(validator_config.SigningKey.PrivateKey));
+    let votingSecretKey =  NimiqPOS.BLSSecretKey.fromHex(validator_config.VotingKey.SecretKey);
+
+    // Create the KeyPairs
+    const validatorKeyPair = Nimiq.KeyPair.derive(validatorPrivateKey);
+    const SigningKeyPair = Nimiq.KeyPair.derive(signingPrivateKey);
+    const votingKeyPair = NimiqPOS.BLSKeyPair.derive(votingSecretKey);
+
+    // Create a new validator registry
+    let validator = new ValidatorRegistry(validatorKeyPair.publicKey.toAddress(), SigningKeyPair.publicKey, votingKeyPair.publicKey);
+
+    // Get the transaction's data
+    let data = validator.get_serialized_data();
+
+    // We monitor the status of the validator registration transactions
+    $.client.addTransactionListener((tx) => {
+        console.log(' Transaction update: ');
+        console.log(tx);
+    }, [recipientAddr]);
+
+    let consensusState = Nimiq.Client.ConsensusState.CONNECTING;
+    $.client.addConsensusChangedListener(async (state) => {
+        consensusState = state;
+        if (state === Nimiq.Client.ConsensusState.ESTABLISHED) {
+            Nimiq.Log.i(TAG, `Blockchain ${config.type}-consensus established in ${(Date.now() - START) / 1000}s.`);
+            const chainHeight = await $.client.getHeadHeight();
+            const chainHeadHash = await $.client.getHeadHash();
+            Nimiq.Log.i(TAG, `Current state: height=${chainHeight}, headHash=${chainHeadHash}`);
+
+            // Once we obtain consensus, we send the validator transactions
+            // Send the txns for validator registration
+            for (let i = 0; i < 6; i++) {
+                console.log('sending transaction: ');
+
+                let transaction = new Nimiq.ExtendedTransaction(validatorKeyPair.publicKey.toAddress(), Nimiq.Account.Type.BASIC, recipientAddr, Nimiq.Account.Type.BASIC, 10, 10, chainHeight, Nimiq.Transaction.Flag.NONE, data[i]);
```

Comment:
I thought the validator is also sending their 100k NIM deposit? This however sends 6 10 luna transactions. Where are the 100k NIM sent?

Reply:
The balance of the account can also be checked by the script in the beginning to make sure it has enough deposit. 

Furthermore, your description said any extra NIM burned will be added as stake. That is also not done by this script, right? The sending of any excess funds?

Reply:
It was decided that these registeration txns do not require any specific value/fee, so I will change the script to send 0 value/0 fee txns. This also means that checking if the sender account has enough funds is no longer needed.

The validator deposit is not paid by this script.

Reply:
Jus realized that 0 value txns are not supported, so I guess we will use 1 LUNA txns for validator registration

Reply:
0 value is not allowed in PoW (and in PoS also only for signalling tx). So must be at least 1 Luna per tx.

Reply:
OK. That means if users generate a fresh validator with this, they need to send it NIM (6 Luna) for the registration to work.

Reply:
Exactly, I will add this check as well